File: _2295.java
import java.io.*;
import java.util.*;

public class _2295 {

    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];

        for(int i=0; i<n; i++){
            arr[i] = Integer.parseInt(br.readLine());
        }

        Arrays.sort(arr);

        ArrayList<Integer> two = new ArrayList<>(n*n);

        for(int i=0; i<n; i++){
            for(int j=0; j<n; j++){
                two.add(arr[i]+arr[j]);
            }
        }
        

        Collections.sort(two);        

        for(int i = n-1; i > -1; i--){
            for(int j = 0; j < n; j++){
                int target = arr[i] - arr[j];

                if(Collections.binarySearch(two, target) >= 0){
                    System.out.println(arr[i]);
                    return;
                }
            }
        }
    }
    
}

File: 2295.py
import sys
from bisect import bisect

n = int(sys.stdin.readline())
arr = []
two = set()

for i in range(n):
    a = int(sys.stdin.readline())
    arr.append(a)

for i in range(n):
    for j in range(n):
        two.add(arr[i]+arr[j])

arr.sort()

found = False
for i in range(n-1, -1, -1):
    if found: break
    for j in range(n):
        target = arr[i] - arr[j]
        
        if target in two:
            print(arr[i])
            found = True
            break
